crates/metrics/src/metrics.rs

```diff
@@ -0,0 +1,49 @@
+#[derive(Debug, Clone, PartialEq)]
+pub struct Metric {
+    pub name: String,
+    pub value: f64,
+}
+
+impl Metric {
+    pub fn new(name: &str, value: f64) -> Self {
+        Self {
+            name: name.to_string(),
+            value,
+        }
+    }
+}
+
+pub trait MetricPublisher: Sync + Send {
+    fn publish(&mut self, metric: &Metric);
```

Comment:
Does this really have to be a mutable self reference? What other kinds of MetricPublishers are you envisioning in the future? The logger one clearly doesn't need the mutable self.

Reply:
@maackle I assumed that mutation is required to send something to aws. I can double check that though.